docs/contributor/benchmarks/otlp-logs-validation.md

```diff
@@ -0,0 +1,312 @@
+# OTel LogPipeline set-up validation
+
+This file documents the process of validating the whole LogPipeline with OTLP output flow. It starts by defining the setup, that consists of the manually deployed log agent, the already-implemented log gateway, and log generators using flog.
+
+The scope is to performance test the agent, observing the resulting values, in terms of throughput, resource consumption, reaction to backpressure, etc. And compare it to the previous FluentBit-based setup.
+
+
+
+## 1. Set-up configuration steps
+
+### With Helm
+
+``` bash
+k apply -f telemetry-manager/config/samples/operator_v1alpha1_telemetry.yaml
+
+// Execute knowledge-hub/scripts/create_cls_log_pipeline.sh with the corresponding environment variables 
+
+helm repo add open-telemetry https://open-telemetry.github.io/opentelemetry-helm-charts
+
+helm install -n kyma-system logging open-telemetry/opentelemetry-collector -f telemetry-manager/docs/contributor/pocs/assets/otel-log-agent-values.yaml
+```
+
+### Manual
+
+``` bash
+k apply -f telemetry-manager/config/samples/operator_v1alpha1_telemetry.yaml
+
+// Execute knowledge-hub/scripts/create_cls_log_pipeline.sh with the corresponding environment variables 
+
+k apply -f ./otlp-logs-validation.yaml
+```
+
+
+
+## 2. Resulting Resources
+
+### Agent ConfigMap (OTel Config)
+
+See [OTLP Logs Validation YAML](./otlp-logs-validation.yaml)
+
+#### Things to take into consideration (at implementation)
+- Dynamically inclusion/exclusion of namespaces, based on LogPipeline spec attributes
+- Exclude FluentBit container in OTel configuration and OTel container in FluentBit configuration
+- `receivers/filelog/operators`: The copy body to `attributes.original` must be avoided if `dropLogRawBody` flag is enabled
+
+### Agent DaemonSet
+
+See [OTLP Logs Validation YAML](./otlp-logs-validation.yaml)
+
+### How does checkpointing work
+
+- By enabling the storeCheckpoint preset (Helm) the `file_storage` extension is activated in the receiver
+- The `file_storage` has the path `/var/lib/otelcol`
+- This is later mounted as a `hostPath` volume in the DaemonSet spec
+- Also set in the `storage` property of the filelog receiver
+
+> `storage` = The ID of a storage extension to be used to store file offsets. File offsets allow the receiver to pick up where it left off in the case of a collector restart. If no storage extension is used, the receiver will manage offsets in memory only.
+
+
+
+## 3. Benchmarking and Performance Tests Results
+
+Setup Configuration:
```

Comment:
These seem to be instructional steps, and probably the order matters. So they should be in an ordered list and with imperative form.

 ```suggestion
1. Apply the configuration:
```